enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,445 @@
+`*.apps.<cluster>.<base>` which all point to the single control-plane node's IP
+address. This allows the installation process and the resulting cluster to
+conveniently work without the user having to think about and configure DNS for
+their cluster (of course external access to the cluster requires the user to
+configure DNS, but this can be done separately after the installation has
+completed).
+
+The issue with those approaches is that they assume that
+`*.apps.<cluster>.<base>` should always point at the single control-plane
+node's IP address. This is of course correct when there's just that single node
+in the cluster, but once you start adding worker nodes to the cluster it starts
+causing a potential problem - the `router-default` deployment created by the
+Cluster Ingress Operator, which is responsible for load balancing ingress
+traffic, targets the "worker" pool using a node selector. As a result, under
+some circumstances, that deployment's pod may eventually find itself running
+also on the newly added worker nodes, as those nodes obviously also belong to
+the worker pool (the reason the control-plane node was also in the worker pool
+is that when during installation there are no worker nodes, the OpenShift
+installer sets the Scheduler CR `.spec.mastersSchedulable` to `true` and as a
+result the control-plane node is in both the "master" and "worker" pools).
+
+This pod floating between the nodes is a problem because user ingress traffic
+has to be directed at the node currently holding the `router-default` pods, and
+since the DNS entries have been "naively" pointed at the original control-plane
+node's IP address (which may no longer be running those pods), ingress traffic
+may no longer work. This can be temporarily solved if DNS is adjusted to point
+at the correct node currently holding the pod or a load-balancer / some virtual
+IP solution is put in place and then the DNS entry can be directed at that
+load-balancer / virtual IP instead of at the node. This enhancement will try
+to prevent this floating altogether so these workarounds will not be needed.
+
+Finally, performing bootstrap-in-place installation of a "none"-platform single
+control-plane node cluster today using the OpenShift Installer does not provide
+the user with a worker node ignition file. As a result, the user is not able to
+manually add worker nodes to the resulting cluster. All current official OCP
+documentation points the user to use "the worker ignition file that get
+generated during installation", but that is not currently generated in such
+bootstrap-in-place installation.
+
+### Goals
+
+- Deal with the "floating ingress" issue encountered in "none"-platform
+single control-plane node clusters which have worker nodes added to them.
+
+- Define how the installer may be modified to generate the worker ignition
+manifest, even when doing bootstrap-in-place "none"-platform single
+control-plane node cluster installation
```

Comment:
I didn't see this fleshed out anywhere, have I missed it?

Reply:
Ah yes, sorry, used to have a TBD for this, looks like it accidentally got deleted during my refactoring work for the enhancement.

Anyway, the installer explicitly avoids generating the `worker.ign` files for BIP installations, this can be trivially fixed by telling it to stop doing that (that == avoiding), I just haven't found the exact piece of code that does that so I didn't mention it in the enhancement, but it should be a trivial matter.